### ISSUE TITLE
RKE2: "Additional Manifest" field containing secrets are visible to all users

### ISSUE THREAD

**Issue by fpants** [rancher/rancher#41527]
We are using Rancher to provision RKE2 clusters with the vSphere provider, and those clusters use the Cisco ACI CNI.
To bootstrap this CNI, we apply its Kubernetes manifests (CRDs, ConfigMaps, Secrets, Deployments, and DaemonSets) using the "Additional Manifest" in the cluster configuration. All the manifests, including Secrets and ConfigMaps containing secrets (API keys in this case), are then visible to all users of the cluster.
This is obviously a security concern for us.

To generalize on this issue: All the manifests set in the spec.rkeConfig.additionalManifest field on the provisioning.cattle.io/v1 Cluster resource results in those manifests containing secrets to be visible to all users of a cluster.
This is the case on all RKE2 clusters regardless of infrastructure provider.

A possible solution to this could be introducing a new field (that isn't visible to normal users) on the Cluster resource with the same purpose as spec.rkeConfig.additionalManifest, allowing us to apply manifests containing secrets.
I think similar solution was made on RKE1 clusters using [this code](https://github.com/rancher/rancher/blob/c349805abca0ba09f5056daeae3bf49da87590ce/pkg/api/norman/store/clustertemplate/store.go).

**Comment by rbreddy** [comment]
The Additional Manifest feature works as expected, and it was not anticipated that users would pass sensitive data there. This is considered a Request for Enhancement

**Comment by bagutzu** [comment]
@nickvth I agree with your point completely. The only issue appears when using the new clustertemplate feature, where we are trying to deploy a cluster with GitOps. If you do this, you will be forced to store secrets in git, in clear.

**Comment by DeyvsonL** [comment]
I support this issue.
I think it would be perfect to have additional manifests visibler only to administrators.
Are there  some guidance on how to set the users/cluster configuration to hide additional manifests from non-administrator users? Or do you have something in the backlog to achieve it?
